tests/ReferenceSqlTest.php

```diff
@@ -350,6 +350,56 @@ public function testOtherAggregates()
         $this->assertNull($l->get('chicken5'));
     }
 
+    public function testReferenceHasOneTraversing()
+    {
+        $a =
+        [
+            'user' => [
+                ['name' => 'Vinny', 'company_id' => 1],
+                ['name' => 'Zoe', 'company_id' => 2],
+            ],
+            'company' => [
+                ['name' => 'Vinny Company'],
+                ['name' => 'Zoe Company'],
+            ],
+            'order' => [
+                ['company_id' => 1, 'description' => 'Vinny Company Order 1', 'amount' => 50.0],
+                ['company_id' => 2, 'description' => 'Zoe Company Order', 'amount' => 10.0],
+                ['company_id' => 1, 'description' => 'Vinny Company Order 2', 'amount' => 15.0],
+            ],
+        ];
+
+        $this->setDb($a);
+
+        $user = (new Model($this->db, 'user'))->addFields(['name', 'company_id']);
+
+        $company = (new Model($this->db, 'company'))->addFields(['name']);
+
+        $user->hasOne('Company', [$company, 'our_field' => 'company_id', 'their_field' => 'id']);
+
+        $order = new Model($this->db, 'order');
+        $order->addField('company_id');
+        $order->addField('description');
+        $order->addField('amount', ['default' => 20]);
+
+        $company->hasMany('Orders', [$order]);
+
+        $user->load(1);
+
+        $this->assertSame([
+            ['id' => '1', 'company_id' => '1', 'description' => 'Vinny Company Order 1', 'amount' => '50'],
+            ['id' => '3', 'company_id' => '1', 'description' => 'Vinny Company Order 2', 'amount' => '15'],
+        ], $user->ref('Company')->ref('Orders')->export());
+
+        $user->unload();
+
+        $this->assertSame([
```

Comment:
so @DarkSide666 was right with his guess yesterday? :)

Reply:
No, I was right. See the updated tests.

Reply:
I see, is cloning on traversing documented?

Reply:
documented here - https://github.com/atk4/data/blame/develop/docs/references.rst#L82

one central clone, that is good, but:
- the method should be named more like `getTheirModelClone()` or `createTheirModelClone`
- is it wanted that when model is created by Closure, the model can be not cloned?

Reply:
The theirModel is generated every time you call `getTheirModel` so the name is correct

Reply:
yes, clone or factory... can be a clone, not a fresh object

Reply:
Cannot be a clone as conditions are assigned to it depending on whether ourModel is loaded or not.

It is good as it is now. I have only solved the issue with referencing.

No need to waste time now on discussion abt method names, etc